pyomo/contrib/doe/tests/test_doe_build.py

```diff
@@ -0,0 +1,478 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright (c) 2008-2024
+#  National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+import json
+from pathlib import Path
+
+from pyomo.common.dependencies import (
+    numpy as np,
+    numpy_available,
+    pandas as pd,
+    pandas_available,
+)
+import pyomo.common.unittest as unittest
+
+from pyomo.contrib.doe import DesignOfExperiments
+from pyomo.contrib.doe.examples.reactor_example import (
+    ReactorExperiment as FullReactorExperiment,
+)
+
+import pyomo.environ as pyo
+
+from pyomo.opt import SolverFactory
+
+ipopt_available = SolverFactory("ipopt").available()
+
+DATA_DIR = Path(__file__).parent
+file_path = DATA_DIR / ".." / "examples" / "result.json"
```

Comment:
In case it's helpful - we have a nifty utility called "this_file_dir":
```
from pyomo.common.fileutils import this_file_dir
currdir = this_file_dir()
```